gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/LocalRegion.java

```diff
@@ -14364,8 +14371,12 @@ private long getEntryOverhead(RegionEntry entry){
 
   protected long calculateEntryOverhead(RegionEntry entry) {
     if (!this.reservedTable() && entryOverHead == -1L && needAccounting()) {
```

Comment:
ensure that entryOverhead is a volatile for double-checked locking to work correctly